pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/ColumnPartitionConfig.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.indexsegment.generator;
+
+import com.linkedin.pinot.core.data.partition.PartitionFunction;
+import com.linkedin.pinot.core.data.partition.PartitionFunctionFactory;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.commons.lang.math.IntRange;
+import org.codehaus.jackson.annotate.JsonIgnore;
+import org.codehaus.jackson.annotate.JsonIgnoreProperties;
+
+
+@SuppressWarnings("unused") // Suppress incorrect warnings as methods used for ser/de.
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ColumnPartitionConfig {
+  public static final String PARTITION_VALUE_DELIMITER = ",";
+
+  String _functionName;
+  String _partitionValues;
+
+  /**
+   * Default constructor for json de-serialization.
+   */
+  public ColumnPartitionConfig() {
+
+  }
+
+  /**
+   * Constructor for the class.
+   *
+   * @param functionName Name of the partition function, includes delimiter separated arguments. For example:
+   *                     "module\t\t20": Function name is module, with argument 20 (divisor).
+   * @param partitionValues Comma separated ranges expected for column values (e.g. [1 2], [3 5], [7 9]).
+   */
+  public ColumnPartitionConfig(@Nonnull String functionName, String partitionValues) {
+    _functionName = functionName;
```

Comment:
Added nonnull for partitionValues. Adding a separate check is flagged as redundant in IDE.